component/otelcol/auth/sigv4/sigv4.go

```diff
@@ -0,0 +1,88 @@
+package sigv4
+
+import (
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/otelcol/auth"
+	"github.com/grafana/agent/pkg/river"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/sigv4authextension"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelconfig "go.opentelemetry.io/collector/config"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "otelcol.auth.sigv4",
+		Args:    Arguments{},
+		Exports: auth.Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := sigv4authextension.NewFactory()
+			return auth.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the otelcol.auth.sigv4 component.
+type Arguments struct {
+	Region     string     `river:"region,attr,optional"`
+	Service    string     `river:"service,attr,optional"`
+	AssumeRole AssumeRole `river:"assume_role,block,optional"`
+}
+
+var _ auth.Arguments = Arguments{}
+
+// Convert implements auth.Arguments.
+func (args Arguments) Convert() otelconfig.Extension {
+	res := sigv4authextension.Config{
+		ExtensionSettings: otelconfig.NewExtensionSettings(otelconfig.NewComponentID("sigv4")),
+		Region:            args.Region,
+		Service:           args.Service,
+		AssumeRole:        *args.AssumeRole.Convert(),
+	}
+	// sigv4authextension.Config has a private member called "credsProvider" which gets initialized when we call Validate().
+	// If we don't call validate, the unit tests for this component will fail.
+	res.Validate()
+	return &res
+}
+
+// Extensions implements auth.Arguments.
+func (args Arguments) Extensions() map[otelconfig.ComponentID]otelcomponent.Extension {
+	return nil
+}
+
+// Exporters implements auth.Arguments.
+func (args Arguments) Exporters() map[otelconfig.DataType]map[otelconfig.ComponentID]otelcomponent.Exporter {
+	return nil
+}
+
+type AssumeRole struct {
+	ARN         string `river:"arn,attr,optional"`
+	SessionName string `river:"session_name,attr,optional"`
+	STSRegion   string `river:"sts_region,attr,optional"`
+}
+
+var _ river.Unmarshaler = (*AssumeRole)(nil)
+
+// UnmarshalRiver implements river.Unmarshaler.
+func (h *AssumeRole) UnmarshalRiver(f func(interface{}) error) error {
+	type assumeRole AssumeRole
+
+	if err := f((*assumeRole)(h)); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
I'm not sure if this is necessary when you're not applying defaults or changing the type we're unmarshaling to. Was there a reason you put this in?

Reply:
Yes, I was not sure of the value this function adds by using `type assumeRole AssumeRole` so I decided to add this just in case. Now when I read your comment, I remembered why we have it - we avoid the infinite loop by calling assumeRole's UnmarshalRiver instead of AssumeRole's UnmarshalRiver. I'll remove the function, thank you!